draft-ietf-taps-impl.md

```diff
@@ -963,3 +963,46 @@ If the application indicates a bound on the expected Send or Receive bitrate, an
 
 * Cost Preferences:
 If the application indicates a preference to avoid expensive paths, and some paths are associated with a monetary cost, an implementation should decrease the ranking of such paths. If the application indicates that it prohibits using expensive paths, paths that are associated with a cost should be purged from the decision tree.
+
+# Reasons for errors {#appendix-reasons-errors}
+
+The Transport Services API {{I-D.ietf-taps-interface}} allows for the seven generic error types to specify a more detailed reason as to why an error occurred. This appendix lists some of the possible reasons.
+
+* BadConfiguration:
+The transport properties and endpoints provided by the application are either contradictory or incomplete. Examples include the lack of a remote endpoint on an active open or using a multicast group address while not requesting a unidirectional receive.
+
+* NoCandidates:
+None of the available transport protocols can satisfy the transport properties provided by the application.
+
+* Unresolvable:
+The remote or local specifier provided by the application can not be resolved.
+
+* CantEstablish:
+The TAPS system was unable to establish a transport-layer connection to the remote endpoint specified by the application.
+
+* PolicyProhibited:
+The system policy prevents the TAPS system from performing the action requested by the application. 
+
+* NotCloneable:
+The protocol stack is not capable of being cloned. 
+
+* MessageToLarge:
+The message size is too big for the taps system to handle.
+
+* ProtocolFailure:
+The underlying protocol stack failed. 
+
+* BadMessageProperties:
+The message properties are either contradictory to the transport properties or they can not be satisfied by the TAPS system. 
+
+* BadReceive:
+The data that was received by the underlying protocol stack could not be fully retrieved by the TAPS system. 
+
+* FailedReceive:
+The underlying protocol stack indicated that the reception of a message has failed.
+
+* ConnectionAborted:
+The connection was aborted by the remote.
+
+* Timeout:
+Delivery of a message to the remote was not possible after a timeout. 
```

Comment:
Nit: "remote endpoint"?

Reply:
I wanted to make it clear that this is the other side of the connection and not the remote endpoint object. There might be a better term for this than "remote".

Reply:
That makes sense.
In this particular case, I think "message delivery" already implies that it's delivery to the other side of the connection, so making that explicit is not necessary here.

However, reading the entire description again, I'm wondering if it makes sense to rephrase it entirely.
How about "Message delivery failed due to a timeout"?